src/docs/cursor-ops.dox

```diff
@@ -161,4 +161,52 @@ that may not be modified or freed by the application.  If a longer scope
 is required, the application must make a copy of the memory before the
 cursor is re-used, closed or reset.
 
+@section cursor_bound Apply bounds to the cursor
+
+A cursor may have a set of bounds applied to it, this restricts the key range
+that the cursor operates within.
+
+Suppose we have a table with keys A to Z. We can restrict the cursor to only operate over
+keys L->Q. If the cursor walks past key Q, the cursor will enter an un-positioned state having
+walked off the "end" of the table from it's perspective. Additionally should the cursor perform a
+search_near it will only search within the key range of L->Q. This can provide a performance boost
+when operating within tables that have a large number of deleted or invisible records. The cursor
+can have a lower bound set or an upper bound set, or both.
+
+```
+cursor->set_key(cursor, A)
+cursor->bound(cursor, bound=lower)
+```
+
+Both or either one of the lower and upper bounds can be set, and the unset bound will be assumed to be off the start or
+end of the key range accordingly. When setting the bounds, the relevant bound is set prior with cursor->set_key.
+This key does not have to exist or be visible in order for bounds setting to succeed.
+
+Cursor operations next, prev, search, search near, modify, insert, update, and remove support bounds. When performing a traversal such as
+cursor->next or cursor->prev, the cursor will only return keys that are within the bounds that have been set. For example, calling cursor->next
+on an unpositioned range bounded cursor will position it on the lower bound rather than the end of the key range.
+
+Example API usage with cursor traversal:
+
+```
+cursor->set_key(cursor, A)
+cursor->bound(cursor, bound=lower,inclusive=false)
+cursor->next(cursor)
+```
```

Comment:
(Optional): The best way to have example code is to put it into one of the programs in examples/c and refer to it here.

Reply:
Thanks for the suggestion! [9b21f18](https://github.com/wiredtiger/wiredtiger/pull/8367/commits/9b21f186bb973ba5c89e96925bf8becfecf64b56)